### DISCUSSION TITLE
Using Directus as a public API for an internal database?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [directus/directus#12772] [Archive]
## Background

We're looking into implementing Directus at our company as a CMS system for our public website. We are interested in the idea of being able to wrap our existing internal company database with a user-friendly admin UI and an API which our public website (SSR on Vercel) could use to grab live data, letting us manage only one single source of truth for our company's data.

## Concern

I spoke today with a team member who manages our current infrastructure on AWS. His main concern was that we wouldn't want an unpredictable flow of public website traffic coming through the Directus API to negatively affect the experience of our current internal users. There are a few way we already know we could reduce this problem:

- On Vercel, send cache headers from our SSR function to move some load out to the edge network and reduce calls.
- In Directus, enable caching to reduce calls to the database.

Combined, these two caching mechanisms would help massively, but we're concerned that it still may not be enough. If enough of either cache invalidates at the same time, a huge spike in traffic could hit our database and cause performance concerns regardless.

## Question

Since our website traffic will be 100% read requests, the ideal solution would be to have public Directus API requests go to a database read replica while internal company users who manage data via the Directus admin UI have their traffic going to the actual database so all CRUD operations work.

Does Directus provide any features to help facilitate such a scenario? Is there a recommended approaching for configuring Directus such that public API requests do not negatively impact the experience of internal company users who rely on the database?

Any help would be much appreciated. We are hoping that we can find a way to make Directus work for us!

**Comment by rijkvanzanten** [comment]
> Does Directus provide any features to help facilitate such a scenario?

Directus doesn't currently have any read replica support 🤔 

> Is there a recommended approaching for configuring Directus such that public API requests do not negatively impact the experience of internal company users who rely on the database?

One easy way is to make sure to configure the access control for the public role as tight as you can. That ensures that public users will never really be able to hurt the database performance too badly. Combine this with the cache, a tight rate limiter (like max 5 requests per second per IP for example), and a CDN in front, and I think you should be more than OK 🙂
